### PR TITLE
Implement named backreferences

### LINKED ISSUE #388: Implement named backreference support
Named backreferences e.g `(?<x>)\k<x>` are not currently supported, but it seems like we ought to be able to lookup the capture index from the capture list, and emit it the same as a numbered backreference (i.e `\1`).

### PR DESCRIPTION
**Pull request by hamishknight**
Use the CaptureList as the source of truth on which index a name corresponds to, and query it when emitting a named backreference.

Resolves #388

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please clean test macOS

**Comment by hamishknight** [comment]
Hmm, removing `namedCaptureOffsets` from MECaptureList appears to be causing seemingly unrelated tests to crash on macOS CI (but not locally for me), any ideas what might be going on there?

**Comment by rxwei** [comment]
@swift-ci please test macOS

**Comment by rxwei** [comment]
looks like a CI glitch!

File: Sources/_StringProcessing/Engine/MEBuilder.swift [hunk 1]
```diff
@@ -38,13 +38,24 @@ extension MEProgram where Input.Element: Hashable {
     // Special addresses or instructions
     var failAddressToken: AddressToken? = nil
 
-    var captureList = CaptureList()
+    private(set) var namedCaptureOffsets: [String: Int] = [:]
```
[the remark this answers is not in the document]
Comment (hamishknight) [reply Sources/_StringProcessing/Engine/MEBuilder.swift]: Makes sense, how does this look?

**Pull request merged**